typescript/sdk/src/token/Token.ts

```diff
@@ -292,6 +314,44 @@ export class Token implements IToken {
       const connection = this.getConnectionForChain(destination);
       assert(connection, `No connection found for chain ${destination}`);
       return this.getIbcAdapter(multiProvider, connection);
+    } else if (standard === TokenStandard.EvmIntent) {
+      assert(
+        this.collateralAddressOrDenom,
+        'collateralAddressOrDenom required for EvmIntent tokens',
+      );
```

Comment:
`addressOrDenom` hosts the address for the Hyperlane7683 instance (router), and `collateralAddressOrDenom` will be the ERC20 token address. That's why it's required when dealing with intents.